hpx/runtime/threads/detail/scheduled_thread_pool_impl.hpp

```diff
@@ -440,7 +440,7 @@ namespace hpx { namespace threads { namespace detail
                 // threads exist or if some other thread has terminated
                 HPX_ASSERT(
                     !sched_->Scheduler::get_thread_count(
-                        unknown, thread_priority_default, thread_num) ||
+                        suspended, thread_priority_default, thread_num) ||
                     sched_->Scheduler::get_state(thread_num) > state_stopping);
```

Comment:
Why is this looking just for `suspended` threads and for all of them as before?

Reply:
Because the exit condition for the scheduling loop checks for `suspended` now (in addition to emptying the work queues). And the exit criteria changed to allow removing/suspending the OS thread on which the removing/suspending HPX thread was started. `suspended` is a sufficient (but slightly overkill) condition to do this, while `unknown` is too much (i.e. can lead to hangups).

The assert could be expanded to check that `work_items_count` and `new_tasks_count` are both 0 as well.

Reply:
That might be a good idea, indeed. Thanks!